src/tools/gcc.lua

```diff
@@ -236,8 +236,20 @@
 
 	gcc.libraryDirectories = {
 		architecture = {
-			x86 = "-L/usr/lib32",
-			x86_64 = "-L/usr/lib64",
+			x86 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib32")
+				end
+				return r
+			end,
+			x86_64 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib64")
+				end
+				return r
+			end,
```

Comment:
Why do these return lists and not strings, like `gcc.ldflags.kind.WindowedApp`? Which would look like this:
```lua
architecture = {
  x86 = function (cfg)
    if cfg.system ~= premake.MACOSX then return "-L/usr/lib32" end
  end,
  x86_64 = function (cfg)
    if cfg.system ~= premake.MACOSX then return "-L/usr/lib64" end
  end,
}
```